demo/nexmark/run.py

```diff
@@ -0,0 +1,134 @@
+    kafka_broker = parser.parse_args().kafka_broker
+
+    # Required files
+    if not os.path.exists(NEXMARK_SQL):
+        raise Exception('Input SQL file {} not found', NEXMARK_SQL)
+
+    # Create program
+    program_name = "nexmark"
+    program_sql = open(NEXMARK_SQL).read()
+    response = requests.put(f"{api_url}/v0/programs/{program_name}", json={
+        "description": "Nexmark benchmark",
+        "code": program_sql
+    })
+    response.raise_for_status()
+    program_version = response.json()["version"]
+
+    # Compile program
+    print(f"Compiling program {program_name} (version: {program_version})...")
+    requests.post(f"{api_url}/v0/programs/{program_name}/compile", json={"version": program_version}).raise_for_status()
+    while True:
+        status = requests.get(f"{api_url}/v0/programs/{program_name}").json()["status"]
+        print(f"Program status: {status}")
+        if status == "Success":
+            break
+        elif status != "Pending" and status != "CompilingRust" and status != "CompilingSql":
+            raise RuntimeError(f"Failed program compilation with status {status}")
+        time.sleep(5)
+
+    # Connectors
+    connectors = []
+    for (connector_name, is_input) in [
+        ("auction", True),
+        ("bid", True),
+        ("person", True),
+        ("q9", False),
+        ("q10", False),
+    ]:
+        name = "kafka_" + ("input" if is_input else "output")
+        json = {
+            "description": "",
+            "config": {
+                "transport": {
+                    "name": name,
+                    "config": {
+                        "auto.offset.reset": "earliest",
+                        "bootstrap.servers": kafka_broker,
+                        "enable.ssl.certificate.verification": "true",
+                        "sasl.mechanism": "PLAIN",
+                        "security.protocol": "PLAINTEXT",
+                    }
+                },
+                "format": {
+                    "name": "csv",
+                    "config": {}
+                }
+            }
+        }
+        config = json["config"]["transport"]["config"]
+        if is_input:
+            config["enable.partition.eof"] = "true"
+            config["topics"] = [connector_name]
+        else:
+            config["topic"] = connector_name
+        requests.put(f"{api_url}/v0/connectors/{connector_name}", json=json).raise_for_status()
+        connectors.append({
+            "connector_name": connector_name,
+            "is_input": is_input,
+            "name": connector_name,
+            "relation_name": connector_name
+        })
+
+    # Create pipeline
+    pipeline_name = "nexmark-pipeline"
+    requests.put(f"{api_url}/v0/pipelines/{pipeline_name}", json={
+        "description": "",
+        "config": {"workers": 6},
+        "program_name": program_name,
+        "connectors": connectors,
+    }).raise_for_status()
+
+    # Stop pipeline
+    print("Stopping pipeline...")
```

Comment:
why is this necessary?
I would make each of these into a little function
they will probably be in a library soon...

Reply:
I added functions.